### ISSUE TITLE
ImagePattern backward compability

### ISSUE THREAD

**Issue by fjonsson** [Python-Markdown/markdown#734]
I am trying to filter image pattern following this suggestion:
https://stackoverflow.com/questions/5930542/check-image-urls-using-python-markdown

The solution worked in version 2.6.11, but after upgrading to 3.0.1 the ImagePattern can no longer be imported.

**Comment by facelessuser** [comment]
When dealing with overrides of existing internal patterns and extensions, there is very little gauruntee that things won't change.  In the release notes for 3.0, we mentioned a new [InlineProcessor](https://python-markdown.github.io/change_log/release-3.0/#added-new-more-flexible-inlineprocessor-class) pattern was introduced, and that has replaced all the internal inline pattern uses. We do still provide the [legacy base classes](https://python-markdown.github.io/extensions/api/#generic-pattern-classes) to help prevent breakage in the transition, but that will only be for a time during transition, so you really should adopt the new InlineProcessor instead.

As far as your specific override, please check out the provided [link](https://github.com/Python-Markdown/markdown/blob/master/markdown/inlinepatterns.py#L595) to see the new pattern that must be overridden. Please reference the documentation and the code to familiarize  yourself with what and how it must be modified.

The biggest thing to note is that the pattern now also accepts the data buffer being processed for more advanced processing, and returns not just the replacement element or text, but also the start and end index in the data buffer that you are replacing.

**Comment by waylan** [comment]
@facelessuser covered the points well. However, the [release notes][1] pointed out the change. For future reference, the release notes are the first place you should look to review changes to a new release.

I'm closing this as there is no actionable item here.

[1]: https://python-markdown.github.io/change_log/release-3.0/#added-new-more-flexible-inlineprocessor-class

**Comment by fjonsson** [comment]
Thanks for the quick answers.

The reason for submitting as a backwards compability bug is the recommended and up-voted solution on stack overflow is no longer valid.

On 12 October 2018 19:18:21 CEST, Waylan Limberg <notifications@github.com> wrote:
>@facelessuser covered the points well. However, the [release notes][1]
>pointed out the change. For future reference, the release notes are the
>first place you should look to review changes to a new release.
>
>I'm closing this as there is no actionable item here.
>
>[1]:
>https://python-markdown.github.io/change_log/release-3.0/#added-new-more-flexible-inlineprocessor-class
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Python-Markdown/markdown/issues/734#issuecomment-429397920

-- 
Skickat från min Android-telefon med K-9 E-post. Ursäkta min fåordighet.

**Comment by facelessuser** [comment]
@fjonsson, That is understood, but 3.0 was released as a major version with the intention of breaking some things.  It is not meant to be backwards compatible with 2.0, though we've provided some paths for certain things temporarily. You are free to update the StackOverflow thread with a recent, updated solution.  Solutions on StackOverflow sometimes become outdated as time goes on -- that cannot be helped.
